social-rs/server/src/main.rs

```diff
@@ -79,38 +81,106 @@ async fn mongo_connect(uri: String) -> Result<Database> {
+    let post_collection: Collection<Post> = db.collection(Post::collection_name());
+
+    // Clean users and posts repositories to begin
+    user_collection.delete_many(doc! {}, None).await?;
+    post_collection.delete_many(doc! {}, None).await?;
+
+    // Show we have empty repositories
+    println!("Repository start state (no text if empty):");
+    let mut user_cursor = user_collection.find(None, None).await?;
+    while user_cursor.advance().await? {
+        println!("{:?}", user_cursor.deserialize_current()?);
+    }
+    let mut post_cusor = post_collection.find(None, None).await?;
+    while post_cusor.advance().await? {
+        println!("{:?}", post_cusor.deserialize_current()?);
+    }
+
 
     let test_user_a = User {
         user_id: Uuid::new_v4(),
-        username: "Alice".to_string(),
+        username: "alice".to_string(),
         first_name: "Alice".to_string(),
-        last_name: "Bob".to_string(),
-        age: 2,
-        email: "alice_bob".to_string(),
+        last_name: "of Wonderland".to_string(),
+        age: 0,
+        email: "alice_of_wonderland@brown.edu".to_string(),
+    };
+
+    let test_user_b = User {
+        user_id: Uuid::new_v4(),
+        username: "bob".to_string(),
+        first_name: "Bob".to_string(),
+        last_name: "Ross".to_string(),
+        age: 0,
+        email: "bob_ross@brown.edu".to_string(),
     };
 
-    let test_post = Post {
+    let post_a = Post {
         text: "Hello this is a post".to_string(),
-        posted_by: test_user_a.user_id,
+        posted_by: test_user_a.user_id.clone(),
         date: "12th March 2023".to_string(),
     };
 
-    println!("Posts collection name: {:?}", Post::collection_name());
-    Post::cascade_delete(&test_post);
+    let post_b = Post {
+        text: "Hello this is another post".to_string(),
+        posted_by: test_user_a.user_id.clone(),
+        date: "12th March 2023".to_string(),
+    };
+
+    let post_c = Post {
+        text: "Hello this is a third post".to_string(),
+        posted_by: test_user_b.user_id.clone(),
+        date: "12th March 2023".to_string(),
+    };
+
+    // add users and posts 
+    user_collection.insert_one(&test_user_a, None).await?;
+    user_collection.insert_one(&test_user_b, None).await?;
+
+    post_collection.insert_one(&post_a, None).await?;
+    post_collection.insert_one(&post_b, None).await?;
+    post_collection.insert_one(&post_c, None).await?;
+
+    println!("New set of users and posts:");
+    let mut user_cursor = user_collection.find(None, None).await?;
+    while user_cursor.advance().await? {
+        println!("{:?}", user_cursor.deserialize_current()?);
+    }
+    let mut post_cusor = post_collection.find(None, None).await?;
+    while post_cusor.advance().await? {
+        println!("{:?}", post_cusor.deserialize_current()?);
+    }
+
 
-    // enforces that the repository i.e. collection is of type User
```

Comment:
I do this above, but with Mongo's `repository` feature vs. collection, which I think simplifies